group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -912,7 +915,9 @@ private void append(
 
                 // If the current write operation is transactional, the current batch
                 // is written before proceeding with it.
-                if (producerId != RecordBatch.NO_PRODUCER_ID) {
+                boolean isTransactional = producerId != RecordBatch.NO_PRODUCER_ID;
```

Comment:
is there a reason we made this a new val? Seems like it is just used below.

Reply:
Nope. Reverted.